src/Tests/Microsoft.NET.TestFramework/TestAsset.cs

```diff
@@ -91,6 +91,28 @@ public TestAsset WithSource()
             return this;
         }
 
+        public TestAsset WithTargetFramework(string targetFramework, string projectName = null)
+        {
+            if (string.IsNullOrEmpty(targetFramework))
+            {
+                return this;
+            }
+
+            return WithProjectChanges((path, project) =>
+            {
+                if (!string.IsNullOrEmpty(projectName))
+                {
+                    if (!projectName.Equals(System.IO.Path.GetFileNameWithoutExtension(path), StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        return;
+                    }
+                }
+
+                var ns = project.Root.Name.Namespace;
+                project.Root.Elements(ns + "PropertyGroup").Elements(ns + "TargetFramework").Single().SetValue(targetFramework);
```

Comment:
>project.Root.Elements(ns + "PropertyGroup").Elements(ns + "TargetFramework").Single( [](start = 16, length = 84)

Consider making this more defensive about project files that do not have a PropertyGroup, or more likely have TargetFrameworks instead of TargetFramework.

Reply:
My thinking here is to keep it simple for the way it's currently being used, but strict so that it will fail obviously if we try to use it in a scenario it doesn't support (e.g. a `TargetFrameworks` property).